source/common/http/conn_manager_impl.cc

```diff
@@ -449,7 +464,14 @@ void ConnectionManagerImpl::onEvent(Network::ConnectionEvent event) {
 
     stats_.named_.downstream_cx_destroy_active_rq_.inc();
     user_agent_.onConnectionDestroy(event, true);
-    resetAllStreams(absl::nullopt);
+    // Note that resetAllStreams() does not actually write anything to the wire. It just resets
+    // all upstream streams and their filter stacks. Thus, there are no issues around recursive
+    // entry.
+    resetAllStreams(response_flag);
```

Comment:
Had a moment where I got worried that resetAllStreams would re-arm the connection duration timer when the last stream was reset, but we're good because you don't just disarm it but clear it above.  might be worth a comment to this next time you or I are in this code but not worth delaying the fix for IMO

Reply:
Sure will add a comment in my next PR.